packages/relay/src/services/singletons/UnirepSocialSynchronizer.ts

```diff
@@ -225,15 +233,68 @@ export class UnirepSocialSynchronizer extends Synchronizer {
             epoch: epoch,
         })
 
-        // if there's no data in EpochKeyAction then do nothing
-        if (rows == 0) return result
-
         // make sure all data whose epochs are before the current ended one is deleted as well
-        db.delete('EpochKeyAction', {
+        if (rows) {
+            db.delete('EpochKeyAction', {
+                where: {
+                    epoch: { lte: epoch },
+                },
+            })
+        }
+
+        // Settle reports
+        const votingReports = await this.db.findMany('ReportHistory', {
             where: {
-                epoch: { lte: epoch },
+                status: ReportStatus.VOTING,
             },
         })
+        // Go through all the voting reports
+        for (const report of votingReports) {
+            const adjudicators = report.adjudicatorsNullifier || []
+            const agreeVotes = adjudicators.filter(
+                (adj: Adjudicator) =>
+                    adj.adjudicateValue === AdjudicateValue.AGREE
+            ).length
+            const disagreeVotes = adjudicators.filter(
+                (adj: Adjudicator) =>
+                    adj.adjudicateValue === AdjudicateValue.DISAGREE
+            ).length
+            // If the current epoch > reportEpoch AND sum of votes > threshold AND vote value > 0
+            if (
+                epoch >= report.reportEpoch &&
+                report.adjudicateCount > REPORT_SETTLE_VOTE_THRESHOLD &&
```

Comment:
should be `report.adjudicateCount >= REPORT_SETTLE_VOTE_THRESHOLD`. if adjudicateCount is equal to 1, it meets the threshold under current configuration.